server/server.go

```diff
@@ -51,17 +51,54 @@ import (
 // registered to it.
 type Server interface {
 	RegisterService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) error
-	GetServiceInfo() *serviceinfo.ServiceInfo
+	GetServiceInfo() map[string]*serviceinfo.ServiceInfo
 	Run() error
 	Stop() error
 }
 
-type server struct {
-	opt     *internal_server.Options
+type service struct {
 	svcInfo *serviceinfo.ServiceInfo
+	handler interface{}
+}
+
+func newService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) service {
+	return service{svcInfo: svcInfo, handler: handler}
+}
+
+type services struct {
+	svcMap     map[string]*service
+	defaultSvc *service
+}
+
+func newServices() *services {
+	return &services{svcMap: map[string]*service{}}
+}
+
+func (s *services) setService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) {
+	svc := newService(svcInfo, handler)
+	if s.defaultSvc == nil {
+		s.defaultSvc = &svc
+	}
+	s.svcMap[svcInfo.ServiceName] = &svc
```

Comment:
no concurrent issue here?

Reply:
@YangruiEmma RegisterService should not be thread-safe, so I think there's no need to lock. (cc. @jayantxie)